src/DocumentFormat.OpenXml/Packaging/OpenXmlPackage.FlatOpc.cs

```diff
@@ -270,5 +294,79 @@ protected static Package FromFlatOpcDocumentCore(XDocument document, Package pac
             package.Flush();
             return package;
         }
+
+        /// <summary>
+        /// Represents the collection of internal package relationships.
+        /// </summary>
+        private class InternalPackageRelationships : IEnumerable<PackageRelationship>
+        {
+            private readonly Package _package;
+
+            /// <summary>
+            /// Initializes a new <see cref="PackageRelationship" /> instance.
+            /// </summary>
+            /// <param name="package">The <see cref="Package" />.</param>
+            public InternalPackageRelationships(Package package)
+            {
+                _package = package ?? throw new ArgumentNullException(nameof(package));
+            }
+
+            /// <inheritdoc />
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return GetEnumerator();
+            }
+
+            /// <inheritdoc />
+            public IEnumerator<PackageRelationship> GetEnumerator()
+            {
+                var relationships = new HashSet<PackageRelationship>(new PackageRelationshipEqualityComparer());
+
+                _package.GetRelationships()
+                    .Where(r => r.TargetMode == TargetMode.Internal)
+                    .ToList()
+                    .ForEach(r => Enumerate(r, relationships));
+
+                return relationships.GetEnumerator();
+            }
+
+#if NET35
+            private void Enumerate(PackageRelationship r, ICollection<PackageRelationship> relationships)
+            {
+                if (relationships.Contains(r))
+                {
+                    return;
+                }
+
+                relationships.Add(r);
+#else
+            private void Enumerate(PackageRelationship r, ISet<PackageRelationship> relationships)
+            {
+                if (!relationships.Add(r))
```

Comment:
The logic here is different. How come you need `ISet<>`? Why not just ask for the HashSet<T> that is generated above? These are all internal, so no worry about leaking details

Reply:
I typically try to use the most abstract types to foster reuse. In this case, that would be `ISet<T>`. But that simply does not exist in `net35` (released 12 years ago).

Reply:
Yes, I agree with trying to do that, but do you *need* it to be `ISet<T>` here? What's the limitation of `ICollection<T>`?

That's the challenge with trying to support `net35` :( We'll probably drop it at some point in the near future, once we can figure out usage and the impact that will have.

Reply:
No, it does not have to be `ISet<T>`. However, if it is not some kind of set, the `Add(T)` method will not return a flag telling me whether the element was added. If it is an `ICollection<T>`, I will have to check for containment first. However, if the underlying collection is a `HashSet<T>`, that will also be O(1). But I need two method calls rather than just one.

As I'm now getting rid of this class, there is no issue anymore. I borrowed it from another use case where net35 is no factor.